www/js/init.js

```diff
@@ -33,8 +33,176 @@
+ * WARNING: Only change these parameters if you know what you are doing
+ */
+// The current version number of this app
+params['appVersion'] = '3.9.1'; // **IMPORTANT** Ensure this is the same as the version number in service-worker.js
+// The PWA server (for use with the browser extensions in ServiceWorker mode)
+params['PWAServer'] = 'https://browser-extension.kiwix.org/current/'; // Include final slash!
+// params['PWAServer'] = 'https://kiwix.github.io/kiwix-js/'; // DEV: Uncomment this line for testing code on GitHub Pages
+// params['PWAServer'] = 'http://localhost:8080/'; // DEV: Uncomment this line (and adjust) for local testing
+// A parameter to determine the Settings Store API in use
+params['storeType'] = getBestAvailableStorageAPI();
+params['hideActiveContentWarning'] = getSetting('hideActiveContentWarning') === 'true';
+params['showUIAnimations'] = getSetting('showUIAnimations') ? getSetting('showUIAnimations') === 'true' : true;
+// Maximum number of article titles to return (range is 5 - 50, default 25)
+params['maxSearchResultsSize'] = getSetting('maxSearchResultsSize') || 25;
+// Turns caching of assets on or off and deletes the cache (it defaults to true unless explicitly turned off in UI)
+params['assetsCache'] = getSetting('assetsCache') !== 'false';
+// Turns caching of the PWA's code on or off and deletes the cache (it defaults to true unless the bypass option is set in Expert Settings)
+params['appCache'] = getSetting('appCache') !== 'false';
+// A parameter to set the app theme and, if necessary, the CSS theme for article content (defaults to 'light')
+params['appTheme'] = getSetting('appTheme') || 'light'; // Currently implemented: light|dark|dark_invert|dark_mwInvert|auto|auto_invert|auto_mwInvert|
+// A global parameter to turn on/off the use of Keyboard HOME Key to focus search bar
+params['useHomeKeyToFocusSearchBar'] = getSetting('useHomeKeyToFocusSearchBar') === 'true';
+// A global parameter to turn on/off opening external links in new tab (for ServiceWorker mode)
+params['openExternalLinksInNewTabs'] = getSetting('openExternalLinksInNewTabs') ? getSetting('openExternalLinksInNewTabs') === 'true' : true;
+// A parameter to disable drag-and-drop
+params['disableDragAndDrop'] = getSetting('disableDragAndDrop') === 'true';
+// A parameter to access the URL of any extension that this app was launched from
+params['referrerExtensionURL'] = getSetting('referrerExtensionURL');
+// A parameter to keep track of the fact that the user has been informed of the switch to SW mode by default
+params['defaultModeChangeAlertDisplayed'] = getSetting('defaultModeChangeAlertDisplayed');
+// A parameter to set the content injection mode ('jquery' or 'serviceworker') used by this app
+params['contentInjectionMode'] = getSetting('contentInjectionMode') ||
+    // Defaults to serviceworker mode when the API is available
+    (isServiceWorkerAvailable() ? 'serviceworker' : 'jquery');
```

Comment:
This method isn't available in init.js. You need to change it to something like `('serviceWorker' in navigator)`. Ditto with `isMessageChannelAvailable`. You can check the code of these methods in app.js, but keep it simple here, just use the basic `x in y` syntax.

Reply:
There are two more methods that are available in app.js but not in init.js namely `setContentInjectionMode` and `switchHomeKeyToFocusSearchBar` but they have quite a good amount of code. Shall I copy them from app.js or move these function calls in app.js only.